### ISSUE TITLE
mask distance and distance threshold

### ISSUE THREAD

**Issue by he-y** [RICE-EIC/Early-Bird-Tickets#2]
Thanks for the great work!
I have two questions.
1. About calculating mask distance. Some layers of VGGNet have different numbers of filters, how to calculate the Hamming distance of the masks with different length? Or I guess it about calculating layer-wise distance?
2. Does the threshold=0.1 hold for different architectures? How to find a proper threshold for a new network?

Thank you!

**Comment by ranery** [comment]
Hi, thanks for asking

1. We regard the model as a whole and flatten each channel to construct the mask, so the dimension of masks remain same during training with fixed model architectures, below is the detailed code for constructing such mask for your reference:
````
mask = torch.zeros(total)
index = 0
for k, m in enumerate(model.modules()):
    if isinstance(m, nn.BatchNorm2d):
        size = m.weight.data.numel()
        weight_copy = m.weight.data.abs().clone()
        _mask = weight_copy.gt(thre.cuda()).float().cuda()
        mask[index:(index+size)] = _mask.view(-1)
        index += size
````
2. We have tried four models (PreResNet-101, VGG-16, ResNet-18/50) on three datasets (CIFAR-10/100 and ImageNet), it generally work for such cases, and the ablation studies regarding the threshold also show that it has great robustness, meaning changing little will not affect the quality of EB tickets much. So I think you can definitely start from 0.1.
